### ISSUE TITLE
create_torrent/generate regression

### ISSUE THREAD

**Issue by claw111** [arvidn/libtorrent#3142]
libtorrent version (or branch): 1.0.11
platform/architecture/compiler:
- Fedora 27/x86_64/gcc 7.3.1 20180303
- Ubuntu 18.04/x86_64/gcc 7.3.0

After much time away from a personal project of mine, I decided to update libtorrent from 1.0.7 to 1.0.11.  After building libtorrent 1.0.11 and updated my project I discovered that the libtorrent library is creating malformed .torrent files.

Specifically, when these created .torrent files are loaded back into libtorrent, libtorrent errors with _missing or invalid 'info' section in torrent file_.

Using the python binding and the following script, I narrowed the regression down to either `create_torrent()` or `generate()` dropping the _info_ section:

```python
import sys
import libtorrent as lt

def _main():
    for filename in sys.argv[1:]:
        with open(filename, 'rb') as thefile:
            decoded = lt.bdecode(thefile.read())
        print repr((filename, decoded, ))

        info = lt.torrent_info(decoded)
        with open(filename + '.result', 'wb') as thefile:
            thefile.write(lt.bencode(
                lt.create_torrent(info).generate()
            ))

if __name__ == '__main__':
    _main()
```

When I run the above script with a well-formed torrent file and libtorrent 1.0.7 then the resulting .torrent file is the same in size as the original .torrent file when the original is provided to the script as arguments.  The  _info_ section is present in the resulting .torrent file as is the original .torrent file.

However, with libtorrent 1.0.11 and the same original .torrent file, the resulting .torrent file is smaller in size to the original and the _info_ section is indeed missing.

**Comment by arvidn** [comment]
are you sure you're using the shared library and python binding from 1.0.11? There was an incompatibility between certain versions at some point, which would have this effect. make sure you're not picking up an older library installed on the system

**Comment by claw111** [comment]
Just a quick note with further observations WRT the _info_ section after `generate()` returns:
1. The _info_ section is present; but
2. The value of the _info_ section is a python `None`, when it is expected to be a dictionary with the same content as the _info_ section from the original .torrent file.

**Comment by arvidn** [comment]
I see. perhaps the problem is that the `preformatted` type can't be round-tripped through python. I'm pretty sure this has been fixed in the current stable branch (1.1.x)

**Comment by arvidn** [comment]
@claw111 if you want to help fix this, I would appreciate a pull request against `RC_1_1` (the branch) extending the python unit test to cover this. (the 1.0.x branch is so old I'm not updating it anymore).

**Comment by arvidn** [comment]
I'm pretty sure this was fixed in 1.1.x. At least there is a test for the python binding to cover (what appears to be) this case
